cloudferrylib/os/identity/keystone.py

```diff
@@ -624,9 +625,14 @@ def _upload_user_tenant_roles(self, user_tenants_roles, users, tenants):
                     role = _role['role']
                     if role['name'] in exists_roles:
```

Comment:
In this case what is the purpose of the `exists_roles` list? I thought this issue should be resolved here...